From 09b64207673464148f8ed28d9a3bf4849638dbcb Mon Sep 17 00:00:00 2001
From: Xiaoyu Yao <xyao@apache.org>
Date: Mon, 11 Jan 2016 16:59:46 -0800
Subject: [PATCH 1254/2848] HDFS-9584. NPE in distcp when ssl configuration
 file does not exist in class path. Contributed by
 Surendra Singh Lilhore.

(cherry picked from commit c2e2e134555010ec28da296bcfef4ba2613a5c6c)
(cherry picked from commit a9cbfd717c3d0cc4e353eff4281df10b5ac3756d)

Change-Id: If94243095b379c27fa1782cfd592ed272cbf5281
---
 .../main/java/org/apache/hadoop/tools/DistCp.java  |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCp.java b/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCp.java
index aec003c..feedd75 100644
--- a/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCp.java
+++ b/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCp.java
@@ -19,6 +19,7 @@
 package org.apache.hadoop.tools;
 
 import java.io.IOException;
+import java.net.URL;
 import java.util.Random;
 
 import org.apache.commons.logging.Log;
@@ -245,8 +246,14 @@ private Job createJob() throws IOException {
    */
   private void setupSSLConfig(Job job) throws IOException  {
     Configuration configuration = job.getConfiguration();
-    Path sslConfigPath = new Path(configuration.
-        getResource(inputOptions.getSslConfigurationFile()).toString());
+    URL sslFileUrl = configuration.getResource(inputOptions
+        .getSslConfigurationFile());
+    if (sslFileUrl == null) {
+      throw new IOException(
+          "Given ssl configuration file doesn't exist in class path : "
+              + inputOptions.getSslConfigurationFile());
+    }
+    Path sslConfigPath = new Path(sslFileUrl.toString());
 
     addSSLFilesToDistCache(job, sslConfigPath);
     configuration.set(DistCpConstants.CONF_LABEL_SSL_CONF, sslConfigPath.getName());
-- 
1.7.9.5

